lib/rubocop/rails/schema_loader.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Rails
+    # It loads db/schema.rb and return Schema object.
+    # Cops refers database schema information with this module.
+    module SchemaLoader
+      extend self
+
+      # It parses `db/schema.rb` and return it.
+      # It returns `nil` if it can't find `db/schema.rb`.
+      # So a cop that uses the loader should handle `nil` properly.
```

Comment:
What do you think about having it raise an exception if `schema.rb` is missing?

Reply:
Personally I think missing `schema.rb` should be handled, it means it shouldn't display a stack trace. Because it is not a bug of rubocop-rails.
But I think it is a good idea to notify missing `shcema.rb` to the user. So, how about a warning instead of an exception?
For example: "`db/schema.rb` is missing. Disable `Rails/UniqueValidationWithoutIndex` cop if you don't use `db/schema.rb`."

Reply:
I don't think there is a need to raise warnings or errors. OTOH, documenting the dependency on db/schema.rb will help users.

Reply:
I do not have a strong opinion for it because I think both of them have pros and cons.
The warning can tell users that the cop does not work. It may be a bug, or using schema.sql instead.
The current behavior conceals it. but it will be easy to use rubocop-rails, because no-configuration is necessary even if the cop doesn't work.



By the way, I updated the documentation to mention existence of db/schema.rb.
https://github.com/rubocop-hq/rubocop-rails/pull/197/files#diff-af4cb6c7e18c12f0a26b2ef45466f731R15